lifecycle-compile.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/sh
+
+mkdir -p /tmp/compile
+mkdir -p /home/vcap/tmp
+mkdir -p /home/vcap/app/.cloudfoundry
+mkdir -p /home/vcap/app/.profile.d
+touch /home/vcap/app/.cloudfoundry/.placeholder
+touch /home/vcap/app/.profile.d/.placeholder
+chown -R vcap:vcap /home/vcap
+
+curl -# -L "https://storage.googleapis.com/golang/go${GO_VERSION}.linux-amd64.tar.gz" | tar -C /tmp/compile -xz
```

Comment:
Whoa.  Interestingly though, github's syntax highlighter (and maybe syntax highlighters in editors) thinks it's a comment, so it read as a potential typo to me... I wonder if we should just use `--progress-bar` instead?  Up to you really!